src/main/java/org/apache/sysml/runtime/codegen/LibSpoofPrimitives.java

```diff
@@ -1914,7 +1915,40 @@ public static void vectGreaterequalAdd(double bval, double[] a, double[] c, int[
 		//invariant to the ordering of inputs
 		return vectLessWrite(b, a, bix, bi, ai, blen, len);
 	}
-	
+
+	//********************* Logical Bitwise Operators ********************//
+
+	/**
+	 * BitwiseAnd Operator
+	 *
+	 * @return
+	 */
+
+	//1. dense vector vs. scalar
+	public static double[] vectBitwandWrite(double[] a, double bval, int ai, int len) {
+		double[] c = allocVector(len, false);
+		for( int j = 0; j < len; j++ )
+			c[j] = (double)( UtilFunctions.toInt(a[ai+j]) & UtilFunctions.toInt(bval) );
```

Comment:
Please use a simple cast to int instead of this machine-precision-aware primitive for consistency with the builtin functions and scalar codegen template.

Reply:
Also please prepare the int value of bval outside the loop.